### PR TITLE
84765 travel claims - enqueue job on timeout

### PR DESCRIPTION
**Pull request by randomsync**
## Summary
This PR adds the functionality to enqueue a job to check the status of travel claim, if the original claim call times out.

- *This work is behind a feature toggle (flipper): YES*

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/84765

## Testing done
- [x] rspecs
- [x] local integration testing

## What areas of the site does it impact?
check_in travel claims

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  Events are being sent to the appropriate logging solution

### CODE REVIEW

File: modules/check_in/app/sidekiq/check_in/travel_claim_submission_worker.rb [hunk 1]
```diff
@@ -57,8 +61,6 @@ def handle_response(opts = {})
                                        [Constants::OH_STATSD_BTSSS_SUCCESS, Constants::OH_SUCCESS_TEMPLATE_ID]
                                      when TravelClaim::Response::CODE_CLAIM_EXISTS
                                        [Constants::OH_STATSD_BTSSS_DUPLICATE, Constants::OH_DUPLICATE_TEMPLATE_ID]
-                                     when TravelClaim::Response::CODE_BTSSS_TIMEOUT
-                                       [Constants::OH_STATSD_BTSSS_TIMEOUT, Constants::OH_TIMEOUT_TEMPLATE_ID]
```
Comment (kanchanasuriya) [inline modules/check_in/app/sidekiq/check_in/travel_claim_submission_worker.rb:61]: Could you please check?  By removing this case statement, we would miss sending out a "`taking longer than usual`" message to the veteran until `check_in_experience_check_claim_status_on_timeout_enabled` is turned on.

Comment (randomsync) [reply modules/check_in/app/sidekiq/check_in/travel_claim_submission_worker.rb:61]: yes, you're right. will fix.

**Review by stevenjcumming — COMMENTED** [review]
Can you move the case statement on `L57` of the `TravelClaimSubmissionWorker` class to a separate method to resolve the rubocop failure for MethodLength? Also consider refactoring to a hash instead of a case statement. 

Example: [https://franzejr.github.io/best-ruby/refactorings/case_with_hashes.html](https://franzejr.github.io/best-ruby/refactorings/case_with_hashes.html)

File: modules/check_in/app/sidekiq/check_in/travel_claim_submission_worker.rb (hunk 1 above)
Comment (randomsync) [reply modules/check_in/app/sidekiq/check_in/travel_claim_submission_worker.rb]: addressed in a516421462a4debd2bd88d08c0bbbf5cfe5d5855

**Comment by randomsync** [comment]
> Can you move the case statement on `L57` of the `TravelClaimSubmissionWorker` class to a separate method to resolve the rubocop failure for MethodLength? Also consider refactoring to a hash instead of a case statement.
> 
> Example: https://franzejr.github.io/best-ruby/refactorings/case_with_hashes.html

addressed in a9123c7

**Review by kanchanasuriya — APPROVED** [review]
LGTM

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**